src/_pytest/python.py

```diff
@@ -152,7 +152,10 @@ def pytest_configure(config):
 def pytest_pyfunc_call(pyfuncitem):
     testfunction = pyfuncitem.obj
     iscoroutinefunction = getattr(inspect, "iscoroutinefunction", None)
-    if iscoroutinefunction is not None and iscoroutinefunction(testfunction):
+    isasyncgenfunction = getattr(inspect, "isasyncgenfunction", None)
```

Comment:
I personally would prefer a `sys.version_info() >= (3, 6)` check, since it's easier to grep for when removing support for old versions, and mypy understands them. Not too important though.

Reply:
sys.version_info is a tuple and not a function

Reply:
Right.